packages/shared/src/hooks/useOnPostClick.ts

```diff
@@ -81,27 +133,53 @@ export default function useOnPostClick({
           await incrementReadingRank();
         }
 
-        if (eventName === 'go to link' && feedQueryKey) {
+        if (eventName === 'go to link') {
           const mutationHandler = () => {
             return {
               read: true,
             };
           };
-          const updateFeedPost = updateCachedPagePost(feedQueryKey, client);
-          const updateFeedPostCache = optimisticPostUpdateInFeed(
-            items,
-            updateFeedPost,
-            mutationHandler,
-          );
-          const postIndex = items.findIndex(
-            (item) => item.type === 'post' && item.post.id === post.id,
-          );
-
-          if (postIndex === -1) {
-            return;
-          }
 
-          updateFeedPostCache({ index: postIndex });
+          if (feedQueryKey) {
+            const updateFeedPost = updateCachedPagePost(feedQueryKey, client);
+            const updateFeedPostCache = optimisticPostUpdateInFeed(
+              items,
+              updateFeedPost,
+              mutationHandler,
+            );
+            const postIndex = items.findIndex(
+              (item) => item.type === 'post' && item.post.id === post.id,
+            );
+
+            if (postIndex === -1) {
+              return;
+            }
+
+            updateFeedPostCache({ index: postIndex });
+          } else if (shouldUseFeedLayoutV1) {
+            const trySetPostRead = (queryKey: QueryKey, id: string) => {
+              const updateFeedPost = updateCachedPagePost(
+                queryKey as unknown[],
+                client,
+              );
+
+              const data = client.getQueryData(
+                queryKey,
+              ) as InfiniteData<FeedData>;
+
+              const { post: foundPost, page, index } = findPost(data, id);
+              if (foundPost) {
+                updateFeedPost(page, index, {
+                  ...foundPost,
+                  ...mutationHandler(),
+                });
+              }
+            };
+
+            getFeedQueryKeys(client).forEach((key) => {
+              trySetPostRead(key, post.id);
+            });
+          }
         }
       },
     // @NOTE see https://dailydotdev.atlassian.net/l/cp/dK9h1zoM
```

Comment:
Right, I need to take a look on the memoization of this. Since the handler is being used on every card, I think it still might make sense to memoize it, but maybe the parameter list needs to be updated 👍

Reply:
Yeah, if you remove the eslint disable it should auto recommend all the dependencies.

Reply:
All the dependencies should now be added.